src/wast-parser.y

```diff
@@ -665,9 +665,9 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
+  | try_check labeling_opt LPAR BLOCK block RPAR catch_list {
```

Comment:
This still seems a bit strange to me. If we use the flat format there is an implicit block, but the folded format syntax requires a block here? Just parsing it, it looks as though there would be an additional block expression in the generated output.